pygmt/src/nearneighbor.py

```diff
@@ -0,0 +1,160 @@
+"""
+nearneighbor - Grid table data using a "Nearest neighbor" algorithm
+"""
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    data_kind,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    E="empty",
+    G="outfile",
+    I="spacing",
+    N="sectors",
+    R="region",
+    S="search_radius",
+    V="verbose",
+    a="aspatial",
+    b="binary",
+    d="nodata",
+    e="find",
+    f="coltypes",
+    g="gap",
+    h="header",
+    i="incols",
+    r="registration",
+    w="wrap",
+)
+@kwargs_to_strings(R="sequence")
+def nearneighbor(x=None, y=None, z=None, data=None, **kwargs):
```

Comment:
Let's go with a data-first and xyz-later approach, following the discussion at https://github.com/GenericMappingTools/pygmt/issues/1479#issuecomment-919636743

```suggestion
def nearneighbor(data=None, x=None, y=None, z=None, **kwargs):
```